include/sdf/Element.hh

```diff
@@ -83,10 +83,20 @@ namespace sdf
     /// \return A copy of this Element.
     public: ElementPtr Clone() const;
 
+    /// \brief Create a copy of this Element.
+    /// \param[out] _errors Vector of errors.
+    /// \return A copy of this Element, NULL if there was an error.
+    public: ElementPtr Clone(sdf::Errors &_errors) const;
+
     /// \brief Copy values from an Element.
     /// \param[in] _elem Element to copy value from.
     public: void Copy(const ElementPtr _elem);
 
+    /// \brief Copy values from an Element.
+    /// \param[in] _elem Element to copy value from.
+    /// \param[out] _errors Vector of errors.
+    public: void Copy(const ElementPtr _elem, sdf::Errors &_errors);
```

Comment:
Since this is targeting `main`, we can change API if we do it this week. For any of these functions that currently return `void`, we could change the return type to `sdf::Errors` to avoid duplicating the functions

Reply:
The idea was to still maintain the old output as an option. If we have `sdf::Errors` as a return and only one function then I believe we would need a way to select one of the two ways of error report. I can think of a boolean flag but doesn't sound very appealing, any other way to solve this?